airbyte-scheduler/src/main/java/io/airbyte/scheduler/WorkerRunFactory.java

```diff
@@ -112,14 +114,13 @@ private WorkerRun createConnectionCheckWorker(JobCheckConnectionConfig config, P
     final StandardCheckConnectionInput checkConnectionInput = getCheckConnectionInput(config);
 
     IntegrationLauncher launcher = createLauncher(config.getDockerImage());
-    DiscoverCatalogWorker discoverCatalogWorker =
-        isAirbyteProtocol(config.getDockerImage()) ? new DefaultDiscoverCatalogWorker(launcher) : new SingerDiscoverCatalogWorker(launcher);
-
+    final CheckConnectionWorker worker =
```

Comment:
singer worker stays the same, but the new one doesn't rely on a discover worker anymore.